support-frontend/app/controllers/RedemptionController.scala

```diff
@@ -0,0 +1,147 @@
+package controllers
+
+import actions.CustomActionBuilders
+import actions.CustomActionBuilders.AuthRequest
+import admin.settings.{AllSettings, AllSettingsProvider}
+import assets.{AssetsResolver, RefPath, StyleContent}
+import cats.data.EitherT
+import cats.implicits._
+import com.gu.identity.model.{User => IdUser}
+import com.gu.support.redemptions.CorporateCustomer
+import com.gu.support.redemptions.redemptions.RedemptionCode
+import controllers.RedemptionController._
+import io.circe.syntax._
+import play.api.libs.circe.Circe
+import play.api.mvc.{AbstractController, Action, AnyContent, ControllerComponents, Result}
+import services.{IdentityService, TestUserService}
+import views.EmptyDiv
+import views.html.helper.CSRF
+import views.html.subscriptionRedemptionForm
+import cats.implicits._
+import lib.RedirectWithEncodedQueryString
+import play.twirl.api.Html
+import views.ViewHelpers.outputJson
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class RedemptionController(
+  val actionRefiners: CustomActionBuilders,
+  val assets: AssetsResolver,
+  settingsProvider: AllSettingsProvider,
+  identityService: IdentityService,
+  testUsers: TestUserService,
+  components: ControllerComponents,
+  fontLoaderBundle: Either[RefPath, StyleContent]
+)(
+  implicit val ec: ExecutionContext
+) extends AbstractController(components) with Circe {
+
+  import actionRefiners._
+
+  implicit val a: AssetsResolver = assets
+
+  implicit val settings: AllSettings = settingsProvider.getAllSettings()
+  val title = "Support the Guardian | Redeem your code"
+  val id = EmptyDiv("subscriptions-redemption-page")
+  val js = "subscriptionsRedemptionPage.js"
+  val css = "digitalSubscriptionCheckoutPage.css" //TODO: Don't need this?
+
+  def displayForm(redemptionCode: String): Action[AnyContent] = NoCacheAction().async {
+    implicit request =>
+      getCorporateCustomer(redemptionCode).value.map(
```

Comment:
if the future is failed, we will get a 500 error.  Failed future would be if the lookup broke, rather than just being a code missing/used.
Not sure if it's any use to recover the future to a success of left, at least with a 5xx we will be notified.

Reply:
👍  we can work this out when we integrate the 2 PRs